src/libraries/Common/tests/TestUtilities/System/Runtime/Intrinsics/ReflectionTester.cs

```diff
@@ -0,0 +1,324 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System.Runtime.Intrinsics;
+
+public sealed class ReflectionTester
```

Comment:
This file seems like a lot of code for testing these scenarios.

I'd expect we already have some of this functionality supported -or- that simply testing the "basic" scenario would be sufficient here.